### PR TITLE
webui: tests: extend the TestInstallationProgress test to include reboot

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

**Review by KKoukiou — COMMENTED** [review]
@martinpitt can you help me with the last remaining test failure here?

The problem is that if they installation fails, and we reboot the machine, the destination disk is unbootable, and the test fails on teardown because of broken ssh connection. 

Is there a way to tell the test to skip teardown maybe?

**Comment by M4rtinK** [comment]
Don't end to end test already support & perform machine reboot & simple booted system testing ? I am a bit confused on which level this testing flow works so that it adds another implementation of the reboot logic. Could it be perhaps shared with what the end-to-end tests use ? :)

**Comment by KKoukiou** [comment]
> Don't end to end test already support & perform machine reboot & simple booted system testing ? I am a bit confused on which level this testing flow works so that it adds another implementation of the reboot logic. Could it be perhaps shared with what the end-to-end tests use ? :)

This is already shared with what e2e tests use. The the helper file moved from the e2e tests direcotyr to the shared folder.

File: ui/webui/test/helpers/utils.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+def add_public_key(machine):
+    with open(machine.identity_file + '.pub', 'r') as pub:
+        public_key = pub.read()
+
+    sysroot_ssh = '/mnt/sysroot/root/.ssh'
+    authorized_keys = os.path.join(sysroot_ssh, 'authorized_keys')
+    machine.execute(fr'''mkdir -p {sysroot_ssh}
+        echo "{public_key}" >> {authorized_keys}
+        chmod 700 {sysroot_ssh}''')
```
Comment (martinpitt) [inline ui/webui/test/helpers/utils.py:28]: Please use:
```
machine.write(authorized_key, public_key, perm="0600")` to write the file. This also creates the parent dir, you can drop the `mkdir`. You may still need the chmod call for the parent dir, though.

**Comment by martinpitt** [comment]
> The problem is that if they installation fails, and we reboot the machine, the destination disk is unbootable, and the test fails on teardown because of broken ssh connection.
> 
> Is there a way to tell the test to skip teardown maybe?

Not right now. You have to modify testlib.py to support some "expect broken machine" flag in MachineCase, which you can set inside a test. Then tearDown() could check that flag and skip all the cleanup.

**Review by KKoukiou — APPROVED** [review]

**Review by KKoukiou — COMMENTED** [review]
@M4rtinK do you have more concerns on this PR apart from the comments which I answered bellow?

**Pull request merged**